### ISSUE TITLE
Non-text elements lack sufficient colour contrast.

### ISSUE THREAD

**Issue by adorayi** [cds-snc/design-gc-conception#2118]
**Description of Issue (required):**

There are meaningful non-text elements found in the solution that have a colour contrast ratio between their foreground and background colours that is below the ratio of 3:1 required by WCAG guidelines.

Example 1: When some of the tabs found throughout the solution (such as on the  the "Card" page) are hovered upon, they are highlighted in a light grey colour (#F1F2F3) which when compared to the white background colour (#FFFFFF) results in a colour contrast of 1.12:1.

Example 2: When some of the navigation links found on every page of  the solution are hovered upon, they are highlighted in a light grey colour (#F1F2F3) which when compared to the white background colour (#FFFFFF) results in a colour contrast of 1.12:1.

Example 3: When some of the navigation links found on every page of  the solution are selected, they are highlighted in a light blue colour (#F1F2F3) which when compared to the white background colour (#FFFFFF) results in a colour contrast of 1.12:1.

Example 4: When an option from dropdown menus throughout the solution (such as the "Date input" page) are hovered upon, they are highlighted in a grey colour (#E1E1E1) which when compared to the white background colour (#FFFFFF) results in a colour contrast ratio of 1.31:1.

Example 5: When the button labelled "Reset controls" marked by an arrow is hovered upon, it is highlighted by a light blue colour (#DBF1FF) which when compared to the white background colour (#FFFFFF) results in a colour contrast ratio of 1.16:1.

**Impact on Users:**
Some users with visual impairments may have trouble seeing non-text elements a colour contrast ratio that is too low.
 
**Sample Code (required):**
<a href="/en/components/card/design/"> Design </a>

**Recommendation (required if not SAMS/OCMC):**
We recommend adjusting the foreground and/or background colours of any affected elements in order to ensure that their contrast ratio meets WCAG standards.

**Additional Resources:**
https://www.w3.org/WAI/WCAG21/Understanding/non-text-contrast.html

**Steps to Reproduce:**
1. Open the solution on the "Card" page: https://design-system.alpha.canada.ca/en/components/card/
2. Take a screenshot of one of the elements in the examples identified in the description and identify its foreground and background colours. 
3. Plug the colours into a colour contrast checker such as: https://coolors.co/contrast-checker

**Actual Result:** 
The colour contrast does not meet the minimum set by WCAG guidelines.

**Expected Result:** 
The colour contrast meets the minimum set by WCAG guidelines.

**Comment by SmartMouthWords** [comment]
Design needs to review to determine colours and contrast

**Comment by nmakuch** [comment]
After doing some reading / research it seems like we're actually ok on these backgrounds or boundaries not hitting 3:1 contrast. 

In the linked W3 article (https://www.w3.org/WAI/WCAG22/Understanding/non-text-contrast.html)
specifically the section on **_Boundaries_**:

"This success criterion does not require that controls have a visual boundary indicating the hit area, but if the visual indicator of the control is the only way to identify the control, then that indicator must have sufficient contrast. If text (or an icon) within a button or placeholder text inside a text input is visible and there is no visual indication of the hit area then the success criterion is passed. If a button with text also has a colored border, since the border does not provide the only indication there is no contrast requirement beyond the text contrast"

The main indicator of the hover state here is not the background fill but the colour change on the text going from black to blue in combination with the border thickness increasing (which all pass WCAG AA or higher). The background fill is supplementary to the text. 

I think we're ok here based on the documentation above, so my suggestion is to leave it as it is and no change is necessary.
